pkg/common/common_test.go

```diff
@@ -200,3 +200,127 @@ func TestGetImagePullPolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestGenerateCliArgsFromConfigMap(t *testing.T) {
+	tests := []struct {
+		name          string
+		cliSubCommand []string
+		configMap     *corev1.ConfigMap
+		expectedArgs  []string
+	}{
+		{
+			name:          "Boolean argument variations",
+			cliSubCommand: []string{"server"},
+			configMap: &corev1.ConfigMap{
+				Data: map[string]string{
+					"--default-snapshot-move":   "true",
+					"--another-key-mix-letters": "TrUe",
+					"key-no-prefix":             "False",
+					"string-not-bool":           "'False'",
+				},
+			},
+			expectedArgs: []string{
+				"server",
+				"--another-key-mix-letters=true",
+				"--default-snapshot-move=true",
+				"--key-no-prefix=false",
+				"--string-not-bool='False'",
+			},
+		},
+		{
+			name:          "All arguments with spaces, some without sigle quotes",
+			cliSubCommand: []string{"server"},
+			configMap: &corev1.ConfigMap{
+				Data: map[string]string{
+					"default-volume-snapshot-locations": "aws:backups-primary, azure:backups-secondary",
+					"log-level":                         "'debug'",
+				},
+			},
+			expectedArgs: []string{
+				"server",
+				"--default-volume-snapshot-locations=aws:backups-primary, azure:backups-secondary",
+				"--log-level='debug'",
+			},
+		},
+		{
+			name:          "Preserve single and double '-' as key prefix",
+			cliSubCommand: []string{"server"},
+			configMap: &corev1.ConfigMap{
+				Data: map[string]string{
+					"-default-volume-snapshot-locations": "aws:backups-primary, azure:backups-secondary",
```

Comment:
We should probably test with "-d" or something with one char here, just because single-dash flags are only valid with single-char short names.

Reply:
So you are suggesting that if the value name:
 1) starts with one `-` it should always be followed with one character
 2) is exactly one character we should append only one `-`

So valid ConfigMap:

```yaml
"-n": "mynamespace"
"n": "mynamespace"

Translates to:
-n=namespace
```

Invalid ConfigMap:
```yaml
"--n": "mynamespace"
```


?

Reply:
@sseago 

I think this one breaks the rule of one dash / one character:

```
  -v, --v Level                          number for the log level verbosity

```

Reply:
lets not validate configmap

I think Scott just suggested a more real life test

Reply:
@mpryc No, I'm not suggesting that we modify the code. one char flag is also a valid full name, so `-` isn't guaranteed to work. I'm just saying that `-foo` will never be valid to pass to velero, while `-f` might be -- so change the test examples to use single char flag name when the example prepends a single `-`.

Reply:
`-v` and `--v` doesn't break what I suggested -- which is "-" is followed by a name of only one char. "--" can be followed by one or more -- the practical reason for this is that the "=" is optional with shortcuts. i.e. `-oyaml` is equivalent to `-o=yaml` -- the only way `-oyaml` works when parsed is that the parser knows that only the first char is the flag shortname, and what follows is the value -- but we're not using that shortcut here, as we're always using the `=` for consistency. That's just a shortcut for users to save keystrokes when typing commands manually.

Reply:
@sseago ok here I did modify a little the logic and now single name characters from the ConfigMap by default are prefixed with single `-`. Longer are prefixed by default with `--`.  Of course user have an option to prefix single character with double `--` if the double `--` are part of the key name in ConfigMap, so the valid are:
```
ConfigMap:
 "-v": "3"
 "--v": 3"
 "v": "3"

The first and last will end up being arg (example for velero server):
$ velero server -v=3

The middle one will end up being:
$ velero server --v=3
```

Reply:
While this won't really make a difference with the current Velero args, since I think `v` is the only one that's the same for both, I don't think it's a safe assumption that if a user enters a single-name arg that it's a shortcut rather than a name. It's possible for an arg to be one char but without a defined shortcut -- and in that case, we're forcing the user to specify name as "--v". I think it's safer to always prepend "--" and document that if the user intends to use the shortcut name instead, they must prepend the single `-`

Reply:
@mpryc ^^

Reply:
@sseago updated, so currently it behaves exactly as you wrote. Thanks.